### PR TITLE
[Prim][PIR] decomp support Inference

### PR DESCRIPTION
**Pull request by cyber-pioneer**
### PR types
New features

### PR changes
Others

### Description
Pcard-66975
Decomp support Inference.
todo: Decomp in inference support op blacklist &whitelist.

### CODE REVIEW

**Review by Charles-hit — APPROVED** [review]

File: test/ir/inference/test_decomp_inference_predictor_run.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    def forward(self, x1, x2):
+        y1 = self.fc1(x1)
+        y2 = self.fc2(x2)
+        y3 = y1 + y2
+        y4 = paddle.nn.functional.layer_norm(y3, y3.shape[1:])
+        z = paddle.nn.functional.softmax(y4)
+        return z
+
+
+class TestPredictorRunWithTensor(unittest.TestCase):
```
Comment (yuanlehome) [inline test/ir/inference/test_decomp_inference_predictor_run.py:40]: 测试取名 应该与decomp相关

File: test/ir/inference/test_decomp_inference_predictor_run.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        pir_predictor = self.init_predictor()
+        pir_output = self.get_inorder_output(pir_predictor)
+        paddle.core._set_prim_all_enabled(False)
+
+        np.testing.assert_allclose(
+            output.numpy().flatten(),
+            pir_output.numpy().flatten(),
+            rtol=1e-6,
+            atol=1e-6,
+        )
```
Comment (yuanlehome) [inline test/ir/inference/test_decomp_inference_predictor_run.py:134]: inorder/disorder在此单测里不需要强调，只需要使用其中一种形式即可

File: paddle/fluid/inference/api/analysis_predictor.cc [hunk 3]
```diff
@@ -786,6 +788,12 @@ bool AnalysisPredictor::PrepareExecutor() {
       pir_program_ = std::move(
           paddle::TranslateLegacyProgramToProgram(*inference_program_));
 
+      if (paddle::prim::PrimCommonUtils::IsFwdPrimEnabled()) {
```
Comment (yuanlehome) [inline paddle/fluid/inference/api/analysis_predictor.cc:791]: 什么时候会enable呢？

File: test/ir/inference/test_decomp_inference_predictor_run.py (hunk 1 above)
Comment (cyber-pioneer) [reply test/ir/inference/test_decomp_inference_predictor_run.py:40]: ok

File: test/ir/inference/test_decomp_inference_predictor_run.py (hunk 2 above)
Comment (cyber-pioneer) [reply test/ir/inference/test_decomp_inference_predictor_run.py:134]: ok

File: paddle/fluid/inference/api/analysis_predictor.cc (hunk 3 above)
Comment (cyber-pioneer) [reply paddle/fluid/inference/api/analysis_predictor.cc:791]: 当设置 FLAGS_prim_forward=True or FLAGS_prim_all=True时开启

**Pull request merged**